src/sentry/web/frontend/setup_wizard.py

```diff
@@ -143,6 +133,80 @@ def get(self, request: HttpRequest, wizard_hash) -> HttpResponseBase:
         context["organizations"] = list(org_mappings_map.values())
         return render_to_response("sentry/setup-wizard.html", context, request)
 
+    def post(self, request: HttpRequest, wizard_hash=None) -> HttpResponse:
+        """
+        This updates the cache content for a specific hash
+        """
+        organization_id = request.POST.get("organizationId", None)
+        project_id = request.POST.get("projectId", None)
+
+        if organization_id is None or project_id is None or wizard_hash is None:
+            return HttpResponseBadRequest()
+
+        try:
+            mapping: OrganizationMapping | None = OrganizationMapping.objects.get(
+                organization_id=organization_id
+            )
+        except OrganizationMapping.DoesNotExist:
+            raise Http404("Organization not found")
```

Comment:
```suggestion
        mapping = get_object_or_404(OrganizationMapping, organization_id=organization_id)
```

There is a nice shortcut/helper function from django https://docs.djangoproject.com/en/5.1/topics/http/shortcuts/#get-object-or-404